### ISSUE TITLE
Custom Nodes documentation

### ISSUE THREAD

**Issue by ecornbower** [blib-la/runpod-worker-comfy#42]
The current documentation recommends forking the entire repo and modifying the `Dockerfile` in order to add custom nodes.

It might be better to recommend creating a new `Dockerfile` and use the pre-build image as a base. For example:

```
FROM timpietruskyblibla/runpod-worker-comfy:2.2.0 as base

WORKDIR /comfyui/custom_nodes

# WAS Node Suite
RUN git clone https://github.com/WASasquatch/was-node-suite-comfyui.git ./was
RUN pip install -r ./was/requirements.txt --no-cache-dir

# Go back to the root
WORKDIR /

# Start the container
CMD /start.sh
```

It's a lot simpler, quicker to build, and easier to update to the latest release.

**Comment by billyberkouwer** [comment]
I've been following this pattern for installing custom nodes and it does work, however I've found this logged when I run the image. Was still fails to load, although it is installed. Do you have any ideas for how to approach a fix for this, Any help would be much appreciated!

```
2024-08-01T22:48:54.605832620Z Import times for custom nodes:
2024-08-01T22:48:54.605854420Z    0.0 seconds: /comfyui/custom_nodes/websocket_image_save.py
2024-08-01T22:48:54.605856540Z    0.0 seconds: /comfyui/custom_nodes/comfyui-various
2024-08-01T22:48:54.605858270Z    0.0 seconds: /comfyui/custom_nodes/ComfyUI-Custom-Scripts
2024-08-01T22:48:54.605869639Z    0.0 seconds: /comfyui/custom_nodes/ComfyUI_essentials
2024-08-01T22:48:54.605871529Z    0.1 seconds: /comfyui/custom_nodes/ComfyUI_FizzNodes
2024-08-01T22:48:54.605883159Z    0.3 seconds: /comfyui/custom_nodes/ComfyUI_Comfyroll_CustomNodes
**2024-08-01T22:48:54.605885169Z    0.8 seconds (IMPORT FAILED): /comfyui/custom_nodes/was-node-suite-comfyui**
```

**Comment by ecornbower** [comment]
@billyberkouwer Sounds like an issue with WAS since your other nodes are working. I would reach out to them instead. Maybe try loading an older version of WAS and see if they broke something.

**Comment by billyberkouwer** [comment]
Thank you!

**Comment by TimPietrusky** [comment]
@ecornbower that sounds like a super good idea! Thank you very much. 

Once #30 is done, we hope that custom nodes can also be added with ease.

**Comment by LotzF** [comment]
@billyberkouwer You're docker container's most likely missing ffmpeg. Docker doesn't have it installed per default and it's not part of the python packages but rather an apt command. Go ahead and install it yourself by altering the apt-get install -y section of the Dockerfile. 

![Screenshot from 2024-09-05 12-21-21](https://github.com/user-attachments/assets/c10e9eb2-e119-4787-9235-a3baa03859c1)

WAS requires ffmpeg to work otherwise the import will faill during startup. 

Cheers
Leo

**Comment by BenDes21** [comment]
> The current documentation recommends forking the entire repo and modifying the `Dockerfile` in order to add custom nodes.
> 
> It might be better to recommend creating a new `Dockerfile` and use the pre-build image as a base. For example:
> 
> ```
> FROM timpietruskyblibla/runpod-worker-comfy:2.2.0 as base
> 
> WORKDIR /comfyui/custom_nodes
> 
> # WAS Node Suite
> RUN git clone https://github.com/WASasquatch/was-node-suite-comfyui.git ./was
> RUN pip install -r ./was/requirements.txt --no-cache-dir
> 
> # Go back to the root
> WORKDIR /
> 
> # Start the container
> CMD /start.sh
> ```
> 
> It's a lot simpler, quicker to build, and easier to update to the latest release.

Is this method still ok ? I cannot use the snapshot file because its contain way too much custom nodes that I dont use for  my specific workflow

**Comment by ecornbower** [comment]
@BenDes21 We ended up migrating away from this library to a custom one, but this approach should still work.

**Comment by TimPietruskyRunPod** [comment]
Custom node installation is now documented at https://github.com/runpod-workers/worker-comfyui/blob/main/docs/customization.md (and \`comfy-node-install\` is the official helper baked into the image — see \`scripts/comfy-node-install.sh\` and the AGENTS.md "Custom node compatibility" section for caveats like dependency version pinning).

The standard pattern:

\`\`\`dockerfile
FROM runpod/worker-comfyui:5.8.5-base
RUN comfy-node-install ComfyUI-Impact-Pack rgthree-comfy
\`\`\`

Closing as documented. If there's something specific the docs don't cover, please open a fresh issue with the gap.
